azure-boards-common/src/main/java/com/synopsys/integration/azure/boards/common/http/AzureBoardsOAuthService.java

```diff
@@ -0,0 +1,86 @@
+package com.synopsys.integration.azure.boards.common.http;
+
+import java.io.IOException;
+import java.net.Proxy;
+import java.util.Optional;
+
+import org.apache.commons.lang3.StringUtils;
+
+import com.google.api.client.auth.oauth2.AuthorizationCodeFlow;
+import com.google.api.client.auth.oauth2.AuthorizationCodeTokenRequest;
+import com.google.api.client.auth.oauth2.BearerToken;
+import com.google.api.client.auth.oauth2.Credential;
+import com.google.api.client.auth.oauth2.StoredCredential;
+import com.google.api.client.auth.oauth2.TokenResponse;
+import com.google.api.client.http.GenericUrl;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.client.util.Base64;
+import com.google.api.client.util.store.DataStoreFactory;
+import com.synopsys.integration.exception.IntegrationException;
+
+public class AzureBoardsOAuthService {
+    private final String authorizationServerUrl;
+    private final Proxy proxy;
+    private final DataStoreFactory dataStoreFactory;
+
+    public AzureBoardsOAuthService(String authorizationServerUrl, Proxy proxy, DataStoreFactory dataStoreFactory) {
+        this.authorizationServerUrl = sanitizeUrl(authorizationServerUrl);
+        this.proxy = proxy;
+        this.dataStoreFactory = dataStoreFactory;
+    }
+
+    public Optional<Credential> getExistingOAuthCredential(AuthorizationCodeFlow authorizationCodeFlow, String userId) throws IntegrationException {
+        try {
+            Credential storedCredential = authorizationCodeFlow.loadCredential(userId);
+            return Optional.ofNullable(storedCredential);
+        } catch (IOException e) {
+            throw new IntegrationException(e);
+        }
+    }
+
+    public String getAuthorizationUrl(AuthorizationCodeFlow authorizationCodeFlow, String redirectUri) {
+        return authorizationCodeFlow.newAuthorizationUrl()
+                   .setClientId(authorizationCodeFlow.getClientId())
+                   .setRedirectUri(redirectUri)
+                   .build();
+    }
+
+    public Credential authorizeAndStoreCredential(AuthorizationCodeFlow authorizationCodeFlow, String userId, String authorizationCode) throws IntegrationException {
+        try {
+            AuthorizationCodeTokenRequest authorizationCodeTokenRequest = authorizationCodeFlow.newTokenRequest(authorizationCode);
+            TokenResponse tokenResponse = authorizationCodeTokenRequest.execute();
+            return authorizationCodeFlow.createAndStoreCredential(tokenResponse, userId);
+        } catch (IOException e) {
+            throw new IntegrationException(e);
+        }
+    }
+
+    private AuthorizationCodeFlow initializeFlow(String clientId) throws IOException {
```

Comment:
Wouldn't this need to be public in the future?  I don't see this method used but the public methods requires an AuthorizationCodeFlow which this private method produces.

Reply:
Yeah, I'm leaving the code in this class alone for now because we will need to determine what gets configured vs gets defaults. I just wanted to leave some code examples in the library, but this may actually become something totally different when we start to implement the channel global config.